components/board/pi/impl/board.go

```diff
@@ -96,7 +96,7 @@ type piPigpio struct {
 	cancelFunc    context.CancelFunc
 	duty          int // added for mutex
 	gpioConfigSet map[int]bool
-	analogReaders map[string]board.Analog
+	analogReaders map[string]*pinwrappers.AnalogSmoother
```

Comment:
What's the goal of this change? By having the struct store interfaces instead of pointers to specific implementations of them, it becomes easier to mock out the analogs during testing, and trivial to change the implementation later. I'd sure expect `pinwrappers.AnalogSmoother` to implement the `board.Analog` interface; why not leave it the way it used to be?

For example, given the changes in this file you could no longer use an `inject.Analog` in a test.

Reply:
The analog reader no longer has a `Close` method. This directly uses the analog smoother to fulfill the analog interface and allow it to Close upon shutdown.

In the numato and pi board, the `Close` method wrapping the analog smoother was returning nil

Reply:
My comment's TL/DR: leave Rand's changes as-is, and remember why it's structured this way.

Reminding myself of the larger picture: 
- The plan is that the `Board` interface will have a `Close()` function that shuts down everything on the board, but specific pins within that board won't, because there shouldn't be a way to close a pin separate from the rest of the board. So, `Close` should be removed from the `Analog` interface.
- The `AnalogSmoother` still needs to have a `Close` on it because it spins up a goroutine for sampling the value. 
- So, the boards that use `AnalogSmoother` are stuck in this limbo, where they need a way to call `Close`, which means they can't store stuff via interface like they used to. This makes it harder to mock out for testing, but frankly mocking things out for testing is often not worth the hassle. This also means that changing the implementation will be more hassle later, but in practice implementations rarely change. We could have avoided this by creating a `CloseableAnalog` interface that encapsulates `Analog` and adds `Close`, but for the moment that's not really worth the hassle, either, despite probably being the textbook approach. That's why the PR is organized the way it is.